core/src/main/clojure/xtdb/xtql/plan.clj

```diff
@@ -474,7 +474,67 @@
 
   ParamRelation
   (plan-query [{:keys [param bindings]}]
-    (plan-rel (plan-expr param) (mapv plan-out-spec bindings))))
+    (plan-rel (plan-expr param) (mapv plan-out-spec bindings)))
+
+  UnionAll
+  (plan-query [{:keys [union-all-queries]}]
+    (reduce (fn [left-plan query]
+              (let [right-plan (plan-query query)]
+                {:ra-plan [:union-all (:ra-plan left-plan) (:ra-plan right-plan)]
+                 :provided-vars (set/union (:provided-vars left-plan) (:provided-vars right-plan))}))
+            (plan-query (first union-all-queries))
+            (rest union-all-queries)))
+
+  Union
+  (plan-query [{:keys [union-queries]}]
+    (reduce (fn [left-plan query]
+              (let [right-plan (plan-query query)]
+                {:ra-plan [:distinct [:union-all (:ra-plan left-plan) (:ra-plan right-plan)]]
+                 :provided-vars (set/union (:provided-vars left-plan) (:provided-vars right-plan))}))
+            (plan-query (first union-queries))
+            (rest union-queries)))
+
+  Intersect
+  (plan-query [{:keys [intersect-queries]}]
+    (reduce (fn [left-plan query]
+              (let [right-plan (plan-query query)]
+                {:ra-plan [:distinct [:intersect (:ra-plan left-plan) (:ra-plan right-plan)]]
```

Comment:
can `distinct` be applied at the outside here? otherwise we're distincting every pair.

(might be what's required - main aim is that it's consistent with SQL (which I know doesn't have variadic intersect 🙂))